pkg/asset/installconfig/ibmcloud/client.go

```diff
@@ -52,29 +55,6 @@ func (e *VPCResourceNotFoundError) Error() string {
 	return "Not Found"
 }
 
-// DNSZoneResponse represents a DNS zone response.
-type DNSZoneResponse struct {
```

Comment:
I actually originally had this in `pkg/types/ibmcloud` and was asked to keep it here. Is there a reason for moving it?

Reply:
Is it because of this?
> Move validation test suite to ibmcloud package

Reply:
I noticed the validation test suite for ibmcloud install config was in it's own package `ibmcloud_test`, which other providers do not do. I figured it will be good to have consistency in how the test files are structured. I attempted to simply update the package name to `ibmcloud`, but ran into `import cycle not allowed in test` error. So these type definitions needed to be moved elsewhere to resolve the issue.

Reply:
Yeah. That's why I ended up putting it in the `ibmcloud_test` package. Otherwise, there would need to be some non-trivial refactoring to avoid the import cycle and keep the types contained within the `ibmcloud` package. As I mentioned in the other comment, there may be a way to do this with mockgen arguments, but I haven't gone too deep into that.

Reply:
I can't seem to find a good way to resolve this when these types are defined in the same package. I've looked into mockgen options, and also simply trying to refactor the code, but it's making it especially difficult because they are used as return types. Was there a reason that these types should not be in `pkg/types/ibmcloud`? Seems like an appropriate place to be.

Reply:
https://github.com/openshift/installer/pull/4923#discussion_r632740396

Reply:
Yes, I came across this comment as well. For time sake, I can just push a commit to revert the refactoring bit.